expression/expression_test.go

```diff
@@ -19,82 +20,137 @@ func TestExpression(t *testing.T) {
-		So(err, ShouldBeNil)
+		expression = &TriggerExpression{MainTargetValue: 40.0, ErrorValue: &errorValue, TriggerType: moira.FallingTrigger}
+		result, err1 = expression.Evaluate()
+		err2 = expression.Validate()
+		So(err1, ShouldBeNil)
+		So(err2, ShouldBeNil)
 		So(result, ShouldResemble, moira.StateOK)
 	})
 
 	Convey("Test Custom", t, func() {
 		expression := "t1 > 10 && t2 > 3 ? ERROR : OK"
-		result, err := (&TriggerExpression{Expression: &expression, MainTargetValue: 11.0, AdditionalTargetsValues: map[string]float64{"t2": 4.0}, TriggerType: moira.ExpressionTrigger}).Evaluate()
-		So(err, ShouldBeNil)
+		trigger := &TriggerExpression{Expression: &expression, MainTargetValue: 11.0, AdditionalTargetsValues: map[string]float64{"t2": 4.0}, TriggerType: moira.ExpressionTrigger}
+		result, err1 := trigger.Evaluate()
+		err2 := trigger.Validate()
+		So(err1, ShouldBeNil)
+		So(err2, ShouldBeNil)
 		So(result, ShouldResemble, moira.StateERROR)
 
 		expression = "min(t1, t2) > 10 ? ERROR : OK"
-		result, err = (&TriggerExpression{Expression: &expression, MainTargetValue: 11.0, AdditionalTargetsValues: map[string]float64{"t2": 4.0}, TriggerType: moira.ExpressionTrigger}).Evaluate()
-		So(err, ShouldResemble, ErrInvalidExpression{fmt.Errorf("functions is forbidden")})
+		trigger = &TriggerExpression{Expression: &expression, MainTargetValue: 11.0, AdditionalTargetsValues: map[string]float64{"t2": 4.0}, TriggerType: moira.ExpressionTrigger}
+		result, err1 = trigger.Evaluate()
+		err2 = trigger.Validate()
+		So(err1, ShouldResemble, ErrInvalidExpression{fmt.Errorf("functions is forbidden")})
+		So(err2, ShouldResemble, ErrInvalidExpression{fmt.Errorf("functions is forbidden")})
 		So(result, ShouldBeEmpty)
 
 		expression = "PREV_STATE"
-		result, err = (&TriggerExpression{Expression: &expression, MainTargetValue: 11.0, AdditionalTargetsValues: map[string]float64{"t2": 4.0}, TriggerType: moira.ExpressionTrigger, PreviousState: moira.StateNODATA}).Evaluate()
-		So(err, ShouldBeNil)
+		trigger = &TriggerExpression{Expression: &expression, MainTargetValue: 11.0, AdditionalTargetsValues: map[string]float64{"t2": 4.0}, TriggerType: moira.ExpressionTrigger, PreviousState: moira.StateNODATA}
+		result, err1 = trigger.Evaluate()
+		err2 = trigger.Validate()
+		So(err1, ShouldBeNil)
+		So(err2, ShouldBeNil)
 		So(result, ShouldResemble, moira.StateNODATA)
 
 		expression = "t1 > 10 && t2 > 3 ? OK : ddd"
-		result, err = (&TriggerExpression{Expression: &expression, MainTargetValue: 11.0, AdditionalTargetsValues: map[string]float64{"t2": 4.0}, TriggerType: moira.ExpressionTrigger}).Evaluate()
-		So(err, ShouldResemble, ErrInvalidExpression{fmt.Errorf("invalid variable value: %w", fmt.Errorf("no value with name ddd"))})
+		trigger = &TriggerExpression{Expression: &expression, MainTargetValue: 11.0, AdditionalTargetsValues: map[string]float64{"t2": 4.0}, TriggerType: moira.ExpressionTrigger}
```

Comment:
The 2 errors look a bit odd, let's keep the one error that comes back from Evaluate, which will either be at runtime from `govaluate` or at compile time from `expr`